src/main/java/com/clickhouse/kafka/connect/sink/processing/Processing.java

```diff
@@ -195,7 +190,8 @@ public void doLogic(List<Record> records) throws IOException, ExecutionException
                         stateProvider.setStateRecord(new StateRecord(topic, partition, rightRangeContainer.getMaxOffset(), rightRangeContainer.getMinOffset(), State.AFTER_PROCESSING));
                         break;
                     case ERROR:
-                        LOGGER.warn(String.format("State mismatch for topic [%s] partition [%s]", topic, partition));
+                        throw new RuntimeException(String.format("State MISMATCH given [%s] records for topic: [%s], partition: [%s], minOffset: [%s], maxOffset: [%s]",
```

Comment:
Will this cause the connector to stop? And then, the user needs to make a manual adjustment to offsets to continue ?